### ISSUE TITLE
escape attribute and <txp:site_url /> tag

### ISSUE THREAD

**Issue by cara-tm** [textpattern/textpattern#1314]
Currently, the `<txp:site_url />` tag do not support the `escape` attribute (see official doc: https://docs.textpattern.io/tags/site_url) which is important especially in a context of json objects such as within JSON-LD.

See the default theme and such as this tag is in use: https://github.com/textpattern/textpattern-default-theme/blob/master/src/templates/pages/default.txp#L38

Textpattern version: 4.7.1

**Comment by bloatware** [comment]
Thanks. It actually supports the global `escape`, but JSON encoding is done with `TEXTPATTERN_JSON` parameter defined as
```
    define('TEXTPATTERN_JSON', JSON_UNESCAPED_UNICODE | JSON_UNESCAPED_SLASHES);
```
I don't know why it is defined this way.

**Comment by bloatware** [comment]
Should be [fixed](https://github.com/textpattern/textpattern/commit/9e558f341c2cbe843d95318f36f0ba799f6536cb)

**Comment by cara-tm** [comment]
Yes, that's perfect Oleg all works fine: `<txp:site_url escape="json" />` renders a well escaped URL!

**Comment by cara-tm** [comment]
Sorry Oleg: it seems do not work properly with `<txp:section url="1" escape="json" />` and `<txp:permlink escape="json" />`.

**Comment by bloatware** [comment]
Both escape slashes in my tests, what's wrong with it?

**Comment by bloatware** [comment]
Could it be that you are on 4.7.2 branch? I have not yet sync it with dev I fear.

**Comment by cara-tm** [comment]
I just replaced the files you change.

**Comment by philwareham** [comment]
Please close this issue if working as expected now. Cheers!
